src/Utils/Extensions.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Stream.Utils
+{
+    internal static class Extensions
+    {
+        internal static void ForEach<T>(this IEnumerable<T> items, Action<T> action)
+        {
+            if ((items == null) || (action == null))
+                return;
+
+            foreach (var item in items)
+                action(item);
+        }
+
+        internal static int SafeCount<T>(this IEnumerable<T> list, int nullCountAs = 0)

```

Comment:
minor, maybe? `internal static int CountOrFallback<T>(this IEnumerable<T> list, int fallbackValue = 0)`. This way it makes more sense that another number can be returned